### DISCUSSION TITLE
Question about PBD algorithms

### DISCUSSION THREAD

**Discussion by zhang-qiang-github** [InteractiveComputerGraphics/PositionBasedDynamics#159] [Q&A]
For example, I have a elastic stick:

![image](https://user-images.githubusercontent.com/29271051/234271302-a9ca8097-8278-4fc6-9b17-913d511c57c3.png)

The left endpoint is fixed (red point), and the right endpoint is free. Then the right endpoint would fall because of the gravity. Let's focus on the right endpoint. The point would fall from z=0 to z=-10, then go up to z=0 if there is no damping.

My **question** is why the point would go up from z=-10 to z=0?

The `PBD` algorithm is:

![image](https://user-images.githubusercontent.com/29271051/234272819-ab622288-b39f-422b-aaf8-f7a27c67587c.png)

Suppose z=-10 before step 5,

step 7: z=-11 because of gravity
step 10: considering the constraint, the z=-11 may go up to z=-9. 

Why the constraint would cause the "go up" (-10 to -9)? 
Why the point don't keep static at z=-10? z=-10 fall to z=-11 because of gravity, and go up from z=-11 to z=-10 because of constraint. Finally, the point keep static at -10.

The "go up" phenomenon is very counterintuitive.

Any suggestion is appreciated~~~~

**Comment by janbender** [comment]
It's the same effect as a spring also does not stop after one simulation step of motion. The constraint causes the point to go up, this yields a new velocity (line 16) which points upwards. In the next simulation step the stick is still not in rest shape, so the constraint will again cause an upward motion plus the velocity pointing up.

**Reply by zhang-qiang-github** [reply]
Thanks~~~
